### DISCUSSION TITLE
How to use .files.getUrl() in view collections

### DISCUSSION THREAD

**Discussion by devpapi0891** [pocketbase/pocketbase#4845] [Q&A]
I created a view table that looks like this

SELECT id, malId, media, details
FROM (
    SELECT id, malId, media, details FROM popular
    UNION
    SELECT id, malId, media, details FROM trending
) AS combined_collections
GROUP BY malId;

And the result 200 looks like this

![image](https://github.com/pocketbase/pocketbase/assets/97945953/5c3a9025-87e3-439d-aaa2-7ffa0f16f1bb)

I am trying to getUrl but the data in media in view collection is only this

![image](https://github.com/pocketbase/pocketbase/assets/97945953/e469e974-cf1b-4c32-ba05-bf995559a7fb)

In which part Im not doing right?

**Comment by ganigeorgiev** [comment]
It is not clear what is the expected result, nor how `files.getUrl` is related to the above.

Columns from subqueries/expressions are currently always evaluated to a `json` field type (_unless there is a `CAST` to number/string_). There is no workaround for this at the moment.

**Reply by devpapi0891** [reply]
I see. Thanks for the info.
